docs/references/apps/Connector.md

```diff
@@ -0,0 +1,156 @@
+# Connector
+
+The Connector app can automatically connect interfaces together. It does this using the query mechanisms inside HELICS to detect all the unconnected interfaces in a cosimulation. Then using given database rules it will establish connections between those interfaces.
+
+It can also run in a two phase mode where in the first mode if federates indicate they have potential interfaces. The connector app will go through the potential interfaces to see if there is a potential connection to be made using the same rules. If a potential interface has a connection available it will send a command to the appropriate interface to go ahead and create those interfaces.
+Then in the second phase it will go ahead an actually make any requested connections from the new and existing unconnected interfaces.
+
+## Connection configuration
+
+The main mechanism to loading connection information is through configuration files.
+
+```text
+#comment line for simple file test
+inp1 pub1 from_to
+```
+
+The simplest is a text file. Comment lines begin with `#`
+the format is
+`<origin> <target> <*direction> <*tags...>` The origin is the interface that is currently unconnected, and target is the interface to connect it to. Direction is optional and assumed bidirectional matching, tags are also optional. Currently only publications, inputs, and endpoints are supported. A more complicated example follows
+
+```text
+#comment line for cascade file test
+# second comment line
+inp1 intermediate1
+intermediate1 intermediate1 bi
+intermediate1 intermediate2 from_to
+intermediate2 intermediate3 from_to
+#comment line in the middle
+intermediate3 pub1 bi
+inp2 intermediate2 from_to
+```
+
+and a short example with tags
+
+```text
+#comment line for simple file test
+inp1 pub1 from_to tag1
+inp2 pub1 tag2 tag3
+```
+
+json format is also supported
+
+```json
+{
+  "connections": [
+    ["inp1", "pub1", "FROM_TO", "tag1"],
+    ["inp2", "pub2", "tag2", "tag3"]
+  ]
+}
+```
+
+## Notes on Tags
```

Comment:
Can tags have any name? How are tags distinguished from the from_to/direction field?

Reply:
If the third field matches a string defining a to_from direction it is assumed to be that otherwise it is a tag